FWCore/Catalog/test/FileLocator_t.cpp

```diff
@@ -2,50 +2,52 @@
 #include "catch2/catch_all.hpp"
 
 #include <array>
+#include <memory>
 #include <string>
+#include <vector>
 
 #include "FWCore/Catalog/interface/FileLocator.h"
 #include "FWCore/Catalog/interface/SiteLocalConfig.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
 #include "FWCore/ServiceRegistry/interface/ServiceRegistry.h"
 #include "FWCore/ServiceRegistry/interface/ServiceToken.h"
 
 #include "TestSiteLocalConfig.h"
 
+namespace {
+  std::unique_ptr<edm::FileLocator> makeFileLocator(edm::ServiceToken const& token, unsigned int iCatalog) {
+    edm::ServiceRegistry::Operate operate(token);
+    edm::Service<edm::SiteLocalConfig> localconfservice;
+    auto const& dataCatalogs = localconfservice->dataCatalogs();
```

Comment:
The `TestSiteLocalConfig` service's data catalogs are defined here
https://github.com/cms-sw/cmssw/blob/ec8fdb3be283f16bd62c4679b366e916114c8648/FWCore/Catalog/test/TestSiteLocalConfig.h#L52-L74
They could be defined in this test source file about as easily.

Reply:
They could except that they are also used in the other unit test, InputFileCatalog_t.cpp...

Reply:
Ah, it was used in the `InputFileCatalog_t.cpp`. I missed that, sorry for the noise.

Reply:
I wonder though if it would be worth of abstracting the creation of the `std::vector<edm::CatalogAttributes>` into a separate function that could then be used here to avoid having to set up the service system just for that. Then the test would also ensure the `FileLocator` implementation stays independent of the `SiteLocalConfig` service.